### PR TITLE
Add option to testimonials shortcode to show full testimonial

### LINKED ISSUE #2814: Testimonial Shortcode: New argument to display full post
The testimonial shortcode only displays an excerpt, never the full post. 

For longer testimonials, this results in a page full of truncated testimonials, with "Read More" links at the end of each. This makes the it impractical to read the page in full, since you need to keep clicking "Read More"  to finish each testimonial, and then click the back button to return to the previous page.

Example page: https://aquariusph.wordpress.com/who-we-are/testimonials-old/

It would give users more flexibility if there were a new argument to specify whether the excerpt or full content should be displayed.

For example: 

`[testimonials use_excerpt=false]`

or

`[testimonials full_content=true]`

Code reference: https://github.com/Automattic/jetpack/blob/69c15d54d3c3772bea0d23d8f7f23793b5c83aab/modules/custom-post-types/testimonial.php#L596

See: https://premium-themes.forums.wordpress.com/topic/testimonials-text-truncated-on-page-using-shortcode

### PR DESCRIPTION
**Pull request by chaselivingston**
Add the `show_excerpt` attribute to the `testimonials` shortcode to give users the option to show the full testimonial. Set to `true` by default.

Fixes #2814

### CODE REVIEW

**Comment by jeherve** [comment]
If we make that change, it might make sense to do the same thing for the Portfolio shortcode as well, to keep things consistent.

File: modules/custom-post-types/testimonial.php [hunk 1]
```diff
@@ -592,8 +593,12 @@ static function jetpack_testimonial_shortcode_html( $atts ) {
...
 						<?php
 						// The content
-						if ( false !== $atts['display_content'] ): ?>
-							<div class="testimonial-entry-content"><?php the_excerpt(); ?></div>
+						if ( false !== $atts['display_content'] ): 
+							if ( true == $atts['show_excerpt'] ) { ?>
+								<div class="testimonial-entry-content"><?php the_excerpt(); ?></div>
+							<?php } else { ?>
+										<div class="testimonial-entry-content"><?php the_content(); ?></div>
+								<?php }
```
Comment (dereksmart) [inline modules/custom-post-types/testimonial.php]: You'll either need to add a closing php tag here, or omit the opening one in the line below

File: modules/custom-post-types/portfolios.php [hunk 2]
```diff
@@ -563,11 +564,13 @@ static function portfolio_shortcode_html( $atts ) {
...
-					<div class="portfolio-entry-content"><?php the_excerpt(); ?></div>
-				<?php endif; ?>
-				</div><!-- close .portfolio-entry -->
-			<?php
+				if ( false !== $atts['display_content'] ): 
++							if ( true == $atts['show_excerpt'] ) { ?>
++								<div class="portfolio-entry-content"><?php the_excerpt(); ?></div><!-- close .portfolio-entry -->
++							<?php } else { ?>
++										<div class="portfolio-entry-content"><?php the_content(); ?></div><!-- close .portfolio-entry -->
++								<?php }
```
Comment (dereksmart) [inline modules/custom-post-types/portfolios.php]: This adds `+`'s in the first character of the above 5 lines

Comment (dereksmart) [inline modules/custom-post-types/portfolios.php]: The spacing of this is pretty messy.  Let's tidy it up

Comment (dereksmart) [inline modules/custom-post-types/portfolios.php]: It seems weird that we're using both `:`/`endif;`, and braces within the same conditional.  Let's stick to one syntax.

**Comment by dereksmart** [comment]
talked with @chaselivingston and we're going with a new argument for display_content rather than adding a whole new attribute

File: modules/custom-post-types/portfolios.php [hunk 3]
```diff
@@ -563,11 +563,13 @@ static function portfolio_shortcode_html( $atts ) {
...
-				<?php endif; ?>
-				</div><!-- close .portfolio-entry -->
-			<?php
+				if ( false !== $atts['display_content'] ) {
+					if ( 'full' === $atts['display_content'] ) {
+						echo '<div class="portfolio-entry-content">' . the_content() . '</div>';
+					} else {
+						echo '<div class="portfolio-entry-content">' . the_excerpt() . '</div>';
+					}
+				}
```
Comment (zinigor) [inline modules/custom-post-types/portfolios.php]: Where does `portfolio-entry` get closed now?

Comment (dereksmart) [inline modules/custom-post-types/portfolios.php]: Looks like it doesn't :/

**Comment by zinigor** [comment]
:+1: Looks good now!

**Comment by danstramer** [comment]
What is the shortcode to display the full content?
Thanks
Dan

**Comment by danstramer** [comment]
Looked at the php file now and saw: display_content=full.
Just didn't see this option documented anywhere.
Dan

**Pull request merged**